src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/FinalizeStep/hooks.tsx

```diff
@@ -0,0 +1,205 @@
+    type,
+    amount,
+    fromDomain,
+    tokenAddress,
+  } = actionData;
+  const { colony } = useColonyContext();
+  const { balances } = colony || {};
+  const { user } = useAppContext();
+
+  const domainBalance = getBalanceForTokenAndDomain(
+    balances,
+    tokenAddress ?? '',
+    Number(nativeMotionDomainId),
+  );
+
+  const requiresDomainFunds: boolean =
+    !!fromDomain &&
+    !!amount &&
+    type !== ColonyActionType.MintTokensMotion &&
+    type !== ColonyActionType.EmitDomainReputationPenaltyMotion &&
+    type !== ColonyActionType.EmitDomainReputationRewardMotion;
+
+  const isFinalizable =
+    !requiresDomainFunds ||
+    // Safe casting since if requiresDomainFunds is true, we know amount is a string
+    BigNumber.from(domainBalance ?? '0').gte(amount as string);
+
+  const transform = mapPayload(
+    () =>
+      ({
+        colonyAddress: colony?.colonyAddress,
+        userAddress: user?.walletAddress,
+        motionId,
+        gasEstimate,
+      } as MotionFinalizePayload),
+  );
+
+  return {
+    transform,
+    isFinalizable,
+  };
+};
+
+export const useClaimConfig = (
+  actionData: MotionAction,
+  startPollingAction: (pollingInterval: number) => void,
+  refetchAction: RefetchAction,
+) => {
+  const {
+    motionData: {
+      stakerRewards,
+      usersStakes,
+      databaseMotionId,
+      remainingStakes,
+    },
+  } = actionData;
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const { pollLockedTokenBalance } = useUserTokenBalanceContext();
+
+  const location = useLocation();
+  const [isClaimed, setIsClaimed] = useState(false);
+
+  const userAddress = user?.walletAddress;
+  const colonyAddress = colony?.colonyAddress;
+  const nativeTokenDecimals = colony?.nativeToken.decimals;
+  const nativeTokenSymbol = colony?.nativeToken.symbol;
+
+  const userStake = usersStakes.find(({ address }) => address === userAddress);
+  const stakerReward = stakerRewards.find(
+    ({ address }) => address === userAddress,
+  );
+
+  // Keep isClaimed state in sync with changes to unclaimed motions on colony object
+  useEffect(() => {
+    if (colony?.motionsWithUnclaimedStakes) {
+      const motionIsUnclaimed = colony.motionsWithUnclaimedStakes.some(
+        ({ motionId }) => motionId === databaseMotionId,
+      );
+
+      if (!motionIsUnclaimed) {
+        setIsClaimed(true);
+        refetchAction();
+      } else {
+        setIsClaimed(false);
+      }
+    }
+  }, [colony?.motionsWithUnclaimedStakes, databaseMotionId, refetchAction]);
+
+  // Keep isClaimed state in sync with user changes
+  useEffect(() => {
+    if (!user) {
+      setIsClaimed(false);
+    } else {
+      setIsClaimed(!!stakerReward?.isClaimed);
+    }
+  }, [user, stakerReward]);
+
+  const {
+    rewards: { yay: yayReward, nay: nayReward },
+    isClaimed: isRewardClaimed,
+  } = stakerReward as StakerRewards;
+
+  if (isRewardClaimed && !isClaimed) {
+    setIsClaimed(true);
+  }
```

Comment:
Don't run side effects inside of render body

Reply:
fixed